File: app/src/androidTest/java/com/byoutline/kickmaterial/robots/UserScenarioRobot.java
package com.byoutline.kickmaterial.robots;

import android.support.test.espresso.contrib.RecyclerViewActions;

import com.byoutline.kickmaterial.R;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static com.byoutline.kickmaterial.espressohelpers.CommonEspressoCalls.onBtnWithIdClick;
import static com.byoutline.kickmaterial.espressohelpers.CommonEspressoCalls.onViewWithTextClick;

public class UserScenarioRobot {

    private static final String DEFALT_PROJECT_NAME = "Smart thermos";
    public static final int ART_CATEGORY_POSITION = 1;

    public UserScenarioResultRobot openDefaultProjectDetails() {
        onViewWithTextClick(DEFALT_PROJECT_NAME);
        return new UserScenarioResultRobot();
    }

    public UserScenarioResultRobot openCategoryScreen() {
        onBtnWithIdClick(R.id.show_categories_fab);
        return new UserScenarioResultRobot();
    }

    public UserScenarioResultRobot selectDefaultCategory() {
        openCategoryScreen();
        onView(withId(R.id.categories_rv)).perform(
                RecyclerViewActions.actionOnItemAtPosition(ART_CATEGORY_POSITION, click()));
        return new UserScenarioResultRobot();
    }

}

File: app/src/androidTest/java/com/byoutline/kickmaterial/robots/UserScenarioRobot.kt
package com.byoutline.kickmaterial.robots

import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.click
import android.support.test.espresso.contrib.RecyclerViewActions
import android.support.test.espresso.matcher.ViewMatchers.withId
import android.support.v7.widget.RecyclerView
import com.byoutline.kickmaterial.R
import com.byoutline.kickmaterial.espressohelpers.onBtnWithIdClick
import com.byoutline.kickmaterial.espressohelpers.onViewWithTextClick

private const val DEFALT_PROJECT_NAME = "Smart thermos"
private const val ART_CATEGORY_POSITION = 1

class UserScenarioRobot {

    fun openDefaultProjectDetails(): UserScenarioResultRobot {
        onViewWithTextClick(DEFALT_PROJECT_NAME)
        return UserScenarioResultRobot()
    }

    fun openCategoryScreen(): UserScenarioResultRobot {
        onBtnWithIdClick(R.id.show_categories_fab)
        return UserScenarioResultRobot()
    }

    fun selectDefaultCategory(): UserScenarioResultRobot {
        openCategoryScreen()
        onView(withId(R.id.categories_rv)).perform(
                RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(ART_CATEGORY_POSITION, click()))
        return UserScenarioResultRobot()
    }
}
